third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/TMEMAlloc1D.cpp

```diff
@@ -21,7 +22,7 @@ class TMEM1DAllocator {
   // Intermediate info to minimize code reuse across functions.
   int numWarps = -1;
   tt::ExpandDimsOp _expandedInput = nullptr;
-  ttng::TMEMAllocOp _allocOp = nullptr;
+  Operation *_allocOp = nullptr;
```

Comment:
This is generalized to allow the reinterpretBuffer. Let me know if there is a better type union syntax.

Reply:
It would be helpful to have a comment stating the possible types and what they represent

Reply:
Will do.